src/Bridge/Symfony/Bundle/DependencyInjection/ApiPlatformExtension.php

```diff
@@ -218,7 +219,8 @@ private function registerMetadataConfiguration(ContainerBuilder $container, arra
 
         $container->getDefinition('api_platform.metadata.extractor.xml')->addArgument($xmlResources);
 
-        if (class_exists(Annotation::class)) {
+        // The "annotation_reader" service is only available if DoctrineBundle is loaded.
+        if (class_exists(Annotation::class) && isset($bundles['DoctrineBundle'])) {
```

Comment:
We should use
```php
$container->hasExtension('doctrine')
```

There is no need to pass the `$bundles`.

Reply:
I mean, they're not strictly the same thing. But we can safely assume that the `doctrine` extension must be registered if `DoctrineBundle` is loaded.

WDYT @dunglas? I think we can simplify the existing code too...

Reply:
To me it's mostly the same thing, except that we check for TwigBundle using `isset($bundles['TwigBundle'])` it made sens to do it the same way for doctrine to keep the code easy to read. But I let @dunglas decide.